### PR TITLE
Add multi platform docker images.

### PR DESCRIPTION
**Pull request by luca-heltai**
I have just finished building a multi-platform dependencies image. 

This PR should enable multi-platform master docker images. 

@gassmoeller @tjhei

### CODE REVIEW

**Review by tjhei — APPROVED** [review]

**Review by gassmoeller — COMMENTED** [review]
The workflow looks good to me, but did you test how long this takes? Github actions has a 6 hour time limit, and it looks like the current x86 build for jammy already takes 2.5 hours (https://github.com/dealii/dealii/actions/runs/9654226039). In the current configuration the arm build will be run on the same runner and may well exceed 6 hours. We may need to look into how to build the different images on different runners and combine them afterwards.

**Comment by marcfehling** [comment]
> In the current configuration the arm build will be run on the same runner and may well exceed 6 hours.

If you add the platforms to the `matrix` logic, there will be a dedicated runner for each plaform. This should help with the time constraints.

**Comment by gassmoeller** [comment]
Yes that should help with the build times, but I think if you build it separately without combining the images afterwards the jobs will also push the images separately, and then you do not have one multi-platform image, but two images for a single platform each. I think the correct way to merge these images afterwards is described [here](https://docs.docker.com/build/ci/github-actions/multi-platform/) but I havent tried it myself.

**Comment by luca-heltai** [comment]
@marcfehling @gassmoeller, thanks for the pointers. This should implement the multi-runner builds.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -50,5 +72,68 @@ jobs:
...
+      
+      - name: Create manifest list and push
+        working-directory: /tmp/digests
+        run: |
+          docker buildx imagetools create $(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
+            $(printf '${{ env.REGISTRY_IMAGE }}@sha256:%s ' *)          
+      
+      - name: Inspect image
+        run: |
+          docker buildx imagetools inspect ${{ env.REGISTRY_IMAGE }}:${{ steps.meta.outputs.version }}
```
Comment (gassmoeller) [inline .github/workflows/docker.yml:139]: github likes to complain about a missing newline at the end of the file

**Review by gassmoeller — APPROVED** [review]
Yes, this looks correct to me (except for the missing newline), but I guess the only way to find out is to merge and see what the runner does.

**Comment by luca-heltai** [comment]
Let's merge and trigger the build then. :)

**Comment by gassmoeller** [comment]
Not sure if there is any point in waiting for the testers, none of them executes the docker workflow. So I am going to merge and hope for the best :-)

**Pull request merged**